### ISSUE TITLE
Some internal binaries are being installed by cmake install and make install.

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#14843]
These libraries are being installed as part of our installation:
```
check_epollexclusive
gen_hpack_tables
gen_legal_metadata_characters
gen_percent_encoding_tables
grpc_cpp_plugin
grpc_create_jwt
grpc_csharp_plugin
grpc_node_plugin
grpc_objective_c_plugin
grpc_php_plugin
grpc_print_google_default_creds_token
grpc_python_plugin
grpc_ruby_plugin
grpc_verify_jwt
```

At least some of the shoudn't be part of our distribution as they are clearly for internal use.
I think the problem is wrong categorization in build.yaml (and/or BUILD file).

We should get rid of these:
```
check_epollexclusive
gen_hpack_tables
gen_legal_metadata_characters
gen_percent_encoding_tables
grpc_create_jwt   # maybe remove?
grpc_print_google_default_creds_token   # maybe remove?
grpc_verify_jwt   # maybe remove?
```

@jboeuf can you comment on the  JWT and creds related binaries?

**Comment by jtattermusch** [comment]
They all seem to be categorized as "tool"  in  build.yaml https://github.com/grpc/grpc/blob/31bdbbeacf84d3a1e867e50b312c9cf46afc20db/build.yaml#L5319

**Comment by jtattermusch** [comment]
CC @matt-kwong   @nicolasnoble

**Comment by jboeuf** [comment]
Yes. The jwt binaries should be considered internal tools and not be part
of the install.

**Comment by vjpai** [comment]
So basically "tool" should not be part of install, but "protoc" should be, correct?

**Comment by jtattermusch** [comment]
Yes, I think so. The name "tool" is a bit misleading.  I think a "tool" is e.g. the grpc_cli client.
The previously mentioned items are "internal tools" or "internal utils". I think that's why we got confused and made them installable.

**Comment by vjpai** [comment]
Strangely, we call "grpc_cli" as "test" which I guess means that it is not installed

**Comment by vjpai** [comment]
Jan, I can't repro this issue for "make". I just tried on a clean VM and did a "make install" and nothing weird is installed. That is also how the template and Makefile look. Do you see this problem for make or just cmake? I think I have the fix in for cmake in #14850 ; testing it now

**Comment by jtattermusch** [comment]
@vjpai   I haven't personally checked the "make" install results, I just thought it would be happening as well - thank for proving this is not the case.

Yes, you are right that grpc_cli  doesn't get installed - we should fix that at some point.
